scripts/ci/openshift.sh

```diff
@@ -25,28 +26,43 @@ scale_worker_nodes() {
     if [[ -n "${ARTIFACT_DIR:-}" ]]; then
         local scale_debug_dir="${ARTIFACT_DIR}/openshift/scaling-debug"
         mkdir -p "${scale_debug_dir}"
-        oc -n openshift-machine-api get machineset -o json > "${scale_debug_dir}/machineset.json"
-        oc -n openshift-machine-api get machines -o json > "${scale_debug_dir}/machines.json"
+        oc -n openshift-machine-api get machineset -o json > "${scale_debug_dir}/initial_machineset.json"
+        oc -n openshift-machine-api get machines -o json > "${scale_debug_dir}/initial_machines.json"
     fi
 
-    original_count="$(get_running_worker_count)"
+    # note: timeout of this loop is handled by the calling context
+    while [[ "$(get_running_worker_count)" != "$expected_initial_worker_count" ]]; do
+        info "Initial machine state does not have the running worker count we expect ($expected_initial_worker_count)"
```

Comment:
I think reporting this as warning to stderr should improve debuggability in OSCI because IIRC stderr items are highlighted with different color.

Let's create something like

```bash
warn() {
    >&2 echo "WARN: $(date): $*"
}
```

in `scripts/lib.sh`.

Then do
```suggestion
        warn "Initial machine state does not have the running worker count we expect ($expected_initial_worker_count). Waiting for machines to become running..."
```

Reply:
It is based on the content of the log, not stdout -v- stderr: https://github.com/openshift/release/blob/e6f6958a6f99d2ef59be6b4302ab4f02b004423a/core-services/prow/02_config/_config.yaml#L76
I'm not sure I agree with adding it here as it'll be the first error that people see when they are looking at OCP results and will not probably be relevant to the failures.

Reply:
Fine to keep it as `stdout`, but my concern is that the user needs to understand what went wrong when the script is killed due to the timeout. My suggestion was an attempt to make it more visible, but certainly it's not the only way.